src/libEth/filters/SubscriptionsImpl.h

```diff
@@ -70,14 +71,32 @@ class SubscriptionsImpl {
     /// Id for websocket dispatch
     Id id = 0;
 
-    /// True if this conn subscribed to pending txns
-    bool subscribedToPendingTxns = false;
+    // populated if this conn subscribed to pending txns.
+    // since a single connection could have multiple subscriptions, we need to
+    // keep a set of subscription IDs here. They are non-overlapping
+    // with subscribedToNewHeads
+    //bool subscribedToPendingTxns = false;
+    std::unordered_set<uint64_t> subscribedToPendingTxn;
```

Comment:
So, the same subscription ID was being used for every connection to newHeads. This is completely fine until there are multiple newHeads subscriptions in the same single connection. This is a bit of a corner case.

Anyway, to solve this, connectinos now keep a set of subscription IDs which are unique and not re-used for subscription types (hewHeads, pendingTxn). This has been confirmed to work with the chainlink software.